### ISSUE TITLE
[BUG] BlobStorage SDK:The download method is unstable and the process is not friendly

### ISSUE THREAD

**Issue by FisherMS** [Azure/azure-sdk-for-net#25316]
**Describe the bug**
1. When downloading, the file size is 0, and the actual size is displayed when the download is complete.
2. When the network is poor, downloading files larger than 100M will become 0 at 100M, and download again.

**Expected behavior**
1. It is hoped that the actual download size can be displayed when downloading.
2. Hope there is a parallel download method to download large files in Blob.Including md5 checksum. I found the deprecated package WindowsAzure.Storage has DownloadRangeToByteArrayAsync method, is there same method in Azure.Storage.Blobs?

**To Reproduce**
Steps to reproduce the behavior
1. Generate SAS Url from a blob container： `https://<REDACTED>.blob.core.windows.net/downdemo/100M.pdf?sp=r&st=2021-11-13T08:01:05Z&se=2022-01-28T16:01:05Z&spr=https&sv=2020-08-04&sr=b&sig=<REDACTED>`
2. Use the code download:
```C#
 BlobClient blobClient = new BlobClient(new Uri(Url));
 blobClient.DownloadTo(@"D:\0\00.pdf");
```

**Environment:**
 - Name and version of the Library package used: Azure.Storage.Blobs 12.10.0.0
 - Hosting platform or OS and .NET runtime version : Win10 + NetCore 3.1 
 - IDE and version : VS 2019

**Comment by amnguye** [comment]
Sorry I'm a little confused on what the bug or issue is, and what the expectation is (this looks like many issues all in one issue)

> When downloading, the file size is 0, and the actual size is displayed when the download is complete.
How big is the blob you're downloading? It might be downloading it in one request, so it is expected that the download starts at 0 then completes to the actual blob size because the block size might be bigger or equal to the size of the file. We only update the file size as the each request comes back successful.

>When the network is poor, downloading files larger than 100M will become 0 at 100M, and download again.
Are you saying that the download resets randomly when the network is poor? I'm confused. This seems like a separate issue than the first one

> It is hoped that the actual download size can be displayed when downloading.
Looks like you're looking for progress tracking feature for download which we have an issue submitted for already 
https://github.com/Azure/azure-sdk-for-net/issues/8506

> Hope there is a parallel download method to download large files in Blob.Including md5 checksum. I found the deprecated package WindowsAzure.Storage has DownloadRangeToByteArrayAsync method, is there same method in Azure.Storage.Blobs?
The download method is parallel, pass the `StorageTransferOptions` object (can also be found in the BlobDownloadToOptions object). 
As for the Content MD5 information please look to our migration guide.
https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/storage/Azure.Storage.Blobs/AzureStorageNetMigrationV12.md#content-hashes

**Comment by Fisher-Joe** [comment]
@amnguye  Thank you .
This download file size is 102M . Sorry ,I found the downloaded size become 0 is because of the servicebus retry .
I will read and test the document you provided today. Thanks again for your work.

**Comment by Fisher-Joe** [comment]
Hi @amnguye , Is there have the block bytes download method?
I want use threads download and split the file different block bytes .

**Comment by amishra-dev** [comment]
Hi @Fisher-Joe,
You can get the blocklist (https://docs.microsoft.com/en-us/dotnet/api/azure.storage.blobs.specialized.blockblobclient.getblocklist?view=azure-dotnet)
And then use the offsets to download the data in parallel threads. @amnguye if we have an existing sample can you please link it here.
